src/main/scala/mesosphere/mesos/TaskBuilder.scala

```diff
@@ -107,9 +108,14 @@ class TaskBuilder(
           s"$numUnusedChecks of its defined health checks."
       )
     }
-
     mesosHealthChecks.headOption.foreach(builder.setHealthCheck)
 
+    val mesosCheck = runSpec.check.collect {
+      case mesosCheck: MesosCheck =>
+        mesosCheck.toMesos(portAssignments)
+    }.flatten
+    mesosCheck.headOption.foreach(builder.setCheck)
```

Comment:
`mesosCheck` is of type `Option[CheckInfo]` so I don't think we need the headOption here

Reply:
What do you think about

```scala
runSpec.check.collect {
  case mesosCheck: MesosCheck =>
    builder.setCheck(mesosCheck.toMesos(portAssignments))
  case _ => ()
}
```

Anyhow, I'm a little confused we have to `flatten` after `collect`. Why is that?